core/trino-main/src/main/java/io/trino/execution/DropTableTask.java

```diff
@@ -86,7 +86,7 @@ public ListenableFuture<Void> execute(
         QualifiedObjectName tableName = redirectionAwareTableHandle.getRedirectedTableName().orElse(originalTableName);
```

Comment:
`tableHandle` and `tableName` are highly correlated.
I'm wondering whether having a `getSchemaTableName()` method in `ConnectorTableHandle` would be worth investigating.

Also the following call seems rather unnecessary because we already provided `originalTableName` while building `RedirectionAwareTableHandle` :

```
QualifiedObjectName tableName = redirectionAwareTableHandle.getRedirectedTableName().orElse(originalTableName);
```

Reply:
```
RedirectionAwareTableHandle redirectionAwareTableHandle = metadata.getRedirectionAwareTableHandle(session, originalTableName);
...
QualifiedObjectName tableName = redirectionAwareTableHandle.getRedirectedTableName().orElse(originalTableName);
```

The `originalTableName` is already known to `RedirectionAwareTableHandle`.

I tried locally to do the refactoring to:

```
QualifiedObjectName tableName = redirectionAwareTableHandle.getRedirectedTableName().orElse(redirectionAwareTableHandle.getTableName());
```
but I don't think is actually a readability improvement.